data-types/timedelta64/README.md

```diff
@@ -0,0 +1,123 @@
+# timedelta64 data type
+
+This document defines a Zarr data type to model the `timedelta64` data type from NumPy. 
+The `timedelta64` data type represents signed temporal durations.
+
+## Background
+
+`timedelta64` is based on a data type with the same name defined in [NumPy](https://NumPy.org/). 
+To provide necessary context, this document first describes how `timedelta64` works in NumPy before 
+detailing its specification in Zarr.
+
+The following references to NumPy are based on version 2.2 of that library.
+
+NumPy defines a data type called `"timedelta64"` to represent signed temporal durations. 
+These durations arise when taking a difference between moments in time. 
+NumPy models moments in time with a related data type called `"datetime64"`. 
+Both data types are described in the [NumPy documentation](https://NumPy.org/doc/stable/reference/arrays.datetime.html), 
+which should be considered authoritative.
+
+`timedelta64` data types are parametrized by a physical unit of duration, like seconds or minutes, 
+and a positive integral scale factor. For example, given a `timedelta64` data type defined with a 
+unit of seconds and a duration 10, the scalar value `1` in that data type represents a duration of 
+10 seconds.   
+
+NumPy represents `timedelta64` scalars with 64 bit signed integers. Negative values are permitted. 
+The smallest 64-bit signed integer, i.e., `-2^63`, represents a non-duration value called 
+"Not a Time", or `NaT`. The `NaT` value serves a role similar to the "Not a Number" value defined in
+some floating point data types. 
+
+### NumPy data type parameters
+
+#### Scale factor
+The NumPy `timedelta64` data type takes a scaling factor. It must be an integer in the range 
+`[1, 2147483647]`, i.e. `[1, 2^31 - 1]`.
+
+While it is possible to construct a NumPy `timedelta64` data type with a scaling factor of `0`, 
+NumPy will automatically normalize this to `1`.
+
+#### Unit
+The NumPy `timedelta64` data type takes a unit parameter, which must be one of the following 
+temporal units:
+
+| Identifier | Meaning     |
+|------------|----------|
+| Y        | year   |
+| M        | month   |
```

Comment:
Just noting that year and month are super problematic as units because they don't actually have a fixed duration (leap years, variable months). I would hate to see us proliferating data with this encoding into the world. But I guess if the goal is numpy compatibility, we should leave them in.

Reply:
100% agree that the numpy definition is problematic. But I think there's value in a data type that numpy users (or zarr v2 users) can adopt without thinking. We should specify a less problematic, more generally useful datetime data type in a separate PR.

Reply:
Maybe it would be useful to rename this data type to `numpy.timedelta64` to signal the intent that it is only meant for compatibility?

Reply:
`numpy.timedelta64` is actually my preferred name, but iirc @rabernat was not a fan.

Reply:
since this naming concern affects all the numpy dtypes, we should resolve that conversation in https://github.com/zarr-developers/zarr-extensions/issues/4.